rapyuta_io_sdk_v2/config.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright 2024 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from dataclasses import dataclass
+
+from rapyuta_io_sdk_v2.constants import (
+    NAMED_ENVIRONMENTS,
+    PROD_ENVIRONMENT_SUBDOMAIN,
+    STAGING_ENVIRONMENT_SUBDOMAIN,
+)
+from rapyuta_io_sdk_v2.exceptions import ValidationError
+
+
+@dataclass
+class Configuration(object):
+    email: str
+    _password: str
+    auth_token: str
+    project_guid: str
+    organization_guid: str
+    environment: str = "ga"  # Default environment is prod
+
+    def __init__(
+        self,
+        project_guid: str = None,
+        organization_guid: str = None,
+        password: str = None,
+        auth_token: str = None,
+        environment: str = None,
+        email: str = None,
+    ):
+        self.email = email
+        self._password = password
+        self.auth_token = auth_token
+        self.project_guid = project_guid
+        self.organization_guid = organization_guid
+        self.environment = environment
+        self.hosts = {}
+        self.set_environment(environment)
+
+    @staticmethod
+    def from_file(file_path: str) -> "Configuration":
+        with open(file_path, "r") as file:
```

Comment:
todo: As discussed earlier this week, copy the logic from click's `get_app_dir` function and use it for loading the config file.